dbtwiz/commands/backfill.py

```diff
@@ -0,0 +1,173 @@
+        full_refresh=full_refresh,
+        service_account=project_config().dbt_service_account,
+        gcp_region=project_config().gcp_region,
+    )
+    with open(YAML_FILE, "w+") as f:
+        f.write(job_spec_yaml)
+    return job_name
+
+
+def job_spec_template():
+    """Templated YAML for Cloud Run job specification"""
+    yaml = """
+    apiVersion: run.googleapis.com/v1
+    kind: Job
+    metadata:
+      name: {{ job_name }}
+      labels:
+        cloud.googleapis.com/location: {{ gcp_region }}
+    spec:
+      template:
+        spec:
+          parallelism: {{ parallelism }}
+          taskCount: {{ task_count }}
+          template:
+            spec:
+              containers:
+              - image: {{ image }}
+                command:
+                - dbtwiz
+                args:
+                - build
+                - --target
+                - "prod"
+                - --select
+                - "{{ selector }}"
+                - --date
+                - "{{ start_date }}"
+                - --use-task-index
+                {% if full_refresh %}
+                - --full-refresh
+                {%- endif %}
+                resources:
+                  limits:
+                    cpu: 1000m
+                    memory: 2Gi
+              maxRetries: 2
+              timeoutSeconds: 900
+              serviceAccountName: {{ service_account }}
+    """
+    return Template(dedent(yaml).lstrip("\n"))
+
+
+def run_command(args: list[str], verbose: bool = False, check: bool = True):
+    """Run the given command in a subprocess"""
+    if verbose:
+        debug(f"Running command: {' '.join(args)}")
+    result = subprocess.run(args)
+    if check:
+        result.check_returncode()
+    return result
+
+
+def backfill(
+        selector: str,
+        first_date: date,
+        last_date: date,
+        full_refresh: bool,
+        parallelism: int,
+        status: bool,
+        verbose: bool
+):
+    job_name = generate_job_spec(
+        selector=selector,
+        date_first=first_date,
+        date_last=last_date,
+        full_refresh=full_refresh,
+        parallelism=parallelism)
+
+    ensure_auth()
+
+    gcp_project = project_config().gcp_project
+    gcp_region = project_config().gcp_region
+
+    info("Preparing job for execution.")
+    run_command(
+        ["gcloud", "run", f"--project={gcp_project}",
+         "jobs", "replace", YAML_FILE],
+        verbose=verbose)
+
+    info("Starting job execution.")
+    run_command(
+        ["gcloud", "run", f"--project={gcp_project}",
+         "jobs", "execute", f"--region={gcp_region}", job_name],
+        verbose=verbose)
+
+    job_url = (
+        f"https://console.cloud.google.com/run/jobs/details"
+        f"/{gcp_region}/{job_name}/executions?project={gcp_project}"
+    )
+    info(f"Job status page: {job_url}")
+    if status:
+        webbrowser.open(job_url)
+        time.sleep(0.5)
+
+
+def halve_str(word: str) -> str:
```

Comment:
Doesn't really matter, but personally I prefer to have functions defined above where they are used

Reply:
Agreed, and this should probably be moved into the support.py file, but we can do that in a separate PR.